cfgov/login/password_validation.py

```diff
@@ -0,0 +1,79 @@
+import functools
+import re
+from datetime import timedelta
+
+from django.contrib.auth import hashers
+from django.core.exceptions import ValidationError
+from django.utils import timezone
+
+
+class PasswordRegexValidator:
+    """Validate that the password matches a given regex"""
+
+    regex = ""
+    message = "Enter a valid password"
+
+    def __init__(self, regex=None, message=None):
+        if regex is not None:
+            self.regex = re.compile(regex)
+        if message is not None:
+            self.message = message
+
+    def validate(self, password, user=None):
+        if not self.regex.search(str(password)):
+            raise ValidationError(self.message)
+
+    def get_help_text(self):
+        return self.message
+
+
+class AgeValidator:
+    """Validate that the password hasn't been changed too recently"""
+
+    def __init__(self, hours):
```

Comment:
```suggestion
    def __init__(self, hours=1):
```

I missed the Django documentation note about [writing your own validators](https://docs.djangoproject.com/en/4.2/topics/auth/passwords/#writing-your-own-validator):

> All constructor arguments should have a default value.